WrocRide/Services/AdminService.cs

```diff
@@ -0,0 +1,199 @@
+
+            return result;
+        }
+
+        public void UpdateDocument(int id, UpdateDocumentDto dto)
+        {
+            var userId = _userContextService.GetUserId;
+            var admin = _dbContext.Admins.FirstOrDefault(u => u.UserId == userId);
+
+            if (admin == null)
+            {
+                throw new NotFoundException("Admin not found");
+            }
+
+            using var dbContextTransaction = _dbContext.Database.BeginTransaction();
+            try
+            {
+                var document = _dbContext.Documents.FirstOrDefault(d => d.Id == id);
+
+                if (document == null)
+                {
+                    throw new NotFoundException("Document not found");
+                }
+
+                document.DocumentStatus = dto.DocumentStatus;
+                document.ExaminationDate = DateTime.UtcNow;
+                document.AdminId = admin.Id;
+
+                if (dto.DocumentStatus == Models.Enums.DocumentStatus.Accepted)
+                {
+                    var status = new UpdateDriverStatusDto
+                    {
+                        DriverStatus = Models.Enums.DriverStatus.Offline
+                    };
+                    var driver = _dbContext.Drivers.FirstOrDefault(d => d.DocumentId == id);
+
+                    if (driver == null)
+                    {
+                        throw new NotFoundException("Driver not found");
+                    }
+
+                    var driverId = driver.Id;
+                    _driverService.UpdateStatus(driverId, status);
+                }
+
+                _dbContext.SaveChanges();
+                dbContextTransaction.Commit();
+            }
+            catch (Exception)
+            {
+                dbContextTransaction.Rollback();
+                throw new Exception();
+            }
+        }
+
+        public DocumentDto GetDocumentByDriverId(int id)
+        {
+            var driver = _dbContext.Drivers.FirstOrDefault(d => d.Id == id);
+
+            if (driver == null)
+            {
+                throw new NotFoundException("Driver not found");
+            }
+
+            var document = _dbContext.Documents.FirstOrDefault(doc => doc.Id == driver.DocumentId);
+
+            if (document == null)
+            {
+                throw new NotFoundException("Document not found");
+            }
+
+            var result = new DocumentDto()
+            {
+                Id = document.Id,
+                DocumentStatus = document.DocumentStatus,
+                FileLocation = document.FileLocation,
+                RequestDate = document.RequestDate
+            };
+
+            return result;
+        }
+        public PagedList<UserDto> GetAll(UserQuery query)
+        {
+            IQueryable<User> baseQuery = _dbContext.Users;
+
+            if (query.RoleId != 0 && query.RoleId != null)
```

Comment:
W sumie to sie zastnawiam o co tu chodzi, bo widzialem ze w validatorze dajesz zakres 0-3 (zamiast 1-3). Rozumimem ze jak jest 0 to masz neutrala i nie sortujesz. To w takim przypadku powiniens zrobic element query.RoleId w UserQuery jako nullable, ze nie trzeba go podawac, a jak sie podaje to z zakresu 1-3.

Reply:
Zastanawiałem się właśnie jak to zrobić. Jak nie podawałem RoleId to postman mi wyrzucał ROleId = 0 i nie widziałem, że to trzeba dodać atrybut nullable.